flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -317,6 +319,10 @@ public AbstractYarnClusterDescriptor createDescriptor(String defaultApplicationN
 		if (cmd.hasOption(SLOTS.getOpt())) {
 			int slots = Integer.valueOf(cmd.getOptionValue(SLOTS.getOpt()));
 			yarnClusterDescriptor.setTaskManagerSlots(slots);
+		} else if (config.containsKey(ConfigConstants.YARN_VCORES)) {
```

Comment:
And in document, there still has introduction of YARN_VCORES.

```
yarn.containers.vcores The number of virtual cores (vcores) per YARN container. By default, the 
number of vcores is set to the number of slots per TaskManager, if set, or to 1, otherwise.
```